unlock-js/src/unlockProvider.js

```diff
@@ -1,11 +1,14 @@
+import { ethers } from 'ethers'
 import { getAccountFromPrivateKey } from './accounts'
 
 // UnlockProvider implements a subset of Web3 provider functionality, sufficient
 // to allow us to use it as a stand-in for MetaMask or other Web3 integration in
 // the browser.
 export default class UnlockProvider {
-  constructor(fallbackProvider) {
-    this.fallbackProvider = fallbackProvider
+  constructor(readOnlyProviderUrl) {
```

Comment:
this looks good. For some consistency with `web3Service` and to enable easy expansion of future props, I'd suggest:

```js
constructo({ readOnlyProvider })
```